pisek/jobs/parts/completeness_check.py

```diff
@@ -52,3 +63,18 @@ def _evaluate(self) -> None:
 
                 if not ok:
                     self._warn(f"{subtask.name} has no dedicated solution")
+
+    def _check_cms_judge(self) -> None:
```

Comment:
Why is this check part of the completeness check? Doesn't it belong in the judge job?

Reply:
It's too early in the judge manager, becaause judge outs are not generated yet.

Reply:
Sure, but the exit code of the judge also isn't checked in the completeness check.

Reply:
Yeah, but you couldn't just print one multilne file that way.

Reply:
I'm not sure what you mean.

Reply:
Well if verbosity <= 0, we only print one multiline judge out.

Reply:
> Sure, but this way the check is only ran if all solutions are tested.

Now runs always.

> Also, is the check ran on the results from chaos monkey?

Added.

Reply:
It still feels out of place in the completeness check. This prevents the check from being cached properly, meaning that all judge outputs have to be opened and examined on every change. However, this might be the best we can do without architectural changes.

Reply:
One random idea I had was that each judge job could check it's own output and store the result. The only responsibility of the completeness check part would be to deduplicate and report the warnings.

Reply:
> It still feels out of place in the completeness check.

I agree. But I don't really have good solution yet.

Reply:
> One random idea I had was that each judge job could check it's own output and store the result. The only responsibility of the completeness check part would be to deduplicate and report the warnings.

But that is not much different than the current version, is it? Plus having this thing cached is going to  save only little time compared to whole run.

Reply:
I don't actually know how long opening and reading so many files will take, it might very well be fine. I just thought it might make sense to read the verdict files only once.

Reply:
> I don't actually know how long opening and reading so many files will take, it might very well be fine.

Tested this on bigger task (2000+ files) and it's basically instant.

> I just thought it might make sense to read the verdict files only once.

We do open solution outputs multiple times btw :innocent: